### PR TITLE
Loading state issue fixed and New Loader animation

### PR DESCRIPTION
**Pull request by Koustavd18**

### CODE REVIEW

**Comment by Koustavd18** [comment]
I have read the CLA Document and I hereby sign the CLA

File: src/assets/customLoader/ParseableAnimate.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import { forwardRef } from 'react';
+import { MantineLoaderComponent } from '@mantine/core';
+
+const ParseableLoader: MantineLoaderComponent = forwardRef(() => (
```
Comment (balaji-jr) [inline src/assets/customLoader/ParseableAnimate.tsx:4]: Why is the forward ref here?

**Review by balaji-jr — CHANGES_REQUESTED** [review]

File: src/assets/customLoader/ParseableLoader.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+// This is a custom Parseable Icon with loading animation
+function ParseableLoader() {
+	return (
+		<MantineProvider
+			theme={{
+				components: {
+					Loader: Loader.extend({
+						defaultProps: {
+							loaders: { ...Loader.defaultLoaders, ring: ParseableAnimate },
+							type: 'ring',
```
Comment (balaji-jr) [inline src/assets/customLoader/ParseableLoader.tsx:13]: Is `ring` a custom loader name ? Or are you just trying to override the defaults?

File: src/pages/Stream/providers/LogsProvider.tsx [hunk 3]
```diff
@@ -601,7 +607,7 @@ const getCleanStoreForRefetch = (store: LogsStore) => {
 		tableOpts: {
 			...tableOpts,
 			pageData: [],
-			totalCount: 0,
+			totalCount: -1,
```
[the remark this answers is not in the document]
Comment (Koustavd18) [reply src/pages/Stream/providers/LogsProvider.tsx:610]: Ok.. Working on that

File: src/pages/Stream/Views/Explore/StaticLogTable.tsx [hunk 4]
```diff
@@ -76,9 +76,13 @@ const TotalLogsCount = () => {
 
 	return (
 		<Stack style={{ alignItems: 'center', justifyContent: 'center', flexDirection: 'row' }} gap={6}>
-			<Text style={{ fontSize: '0.7rem' }}>{`Showing ${showingCount} out of`}</Text>
-			<TotalCount totalCount={totalCount} />
-			<Text style={{ fontSize: '0.7rem' }}>records</Text>
+			{totalCount < 0 ? null : (
```
[the remark this answers is not in the document]
Comment (Koustavd18) [reply src/pages/Stream/Views/Explore/StaticLogTable.tsx:79]: 👍🏻

File: src/assets/customLoader/ParseableLoader.tsx [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import { MantineProvider, Loader } from '@mantine/core';
+import ParseableAnimate from './ParseableAnimate';
+
+// This is a custom Parseable Icon with loading animation
+function ParseableLoader() {
+	return (
+		<MantineProvider
```
[the remark this answers is not in the document]
Comment (Koustavd18) [reply src/assets/customLoader/ParseableLoader.tsx:7]: This parseable logo animation built using mantine theme. No other theme provider is used

File: src/assets/customLoader/ParseableLoader.tsx (hunk 2 above)
Comment (Koustavd18) [reply src/assets/customLoader/ParseableLoader.tsx:13]: Working on that

Comment (Koustavd18) [reply src/assets/customLoader/ParseableLoader.tsx:13]: Changed.. Not overriding the ring loader

File: src/assets/customLoader/ParseableLoader.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (balaji-jr) [reply src/assets/customLoader/ParseableLoader.tsx:7]: This still needs to be done.

File: src/pages/Stream/Views/Explore/StaticLogTable.tsx [hunk 6]
```diff
@@ -240,7 +249,7 @@ const Footer = (props: { loaded: boolean }) => {
 	return (
 		<Stack className={tableStyles.footerContainer} gap={0}>
 			<Stack w="100%" justify="center" align="flex-start">
-				{!props.loaded ? null : <TotalLogsCount />}
+				{!props.loaded ? <Loader type="dots" /> : <TotalLogsCount />}
```
Comment (balaji-jr) [inline src/pages/Stream/Views/Explore/StaticLogTable.tsx:252]: This still takes the loading state of the table data. You should implement a separate loading state on fetchQueryCount.

**Review by balaji-jr — CHANGES_REQUESTED** [review]

File: src/pages/Stream/Views/Explore/StaticLogTable.tsx [hunk 7]
```diff
@@ -191,13 +200,20 @@ const ErrorView = (props: { message: string }) => {
...
+				alignItems: 'center',
+				justifyContent: 'center',
+			}}>
+			<Loader type="parseable" />
 		</Stack>
 	);
 };
 
-const Footer = (props: { loaded: boolean }) => {
+const Footer = (props: { loaded: boolean; isLoading: boolean; hasNoData: boolean }) => {
```
Comment (balaji-jr) [inline src/pages/Stream/Views/Explore/StaticLogTable.tsx:216]: Can you rename the props with more appropriate names ?

**Review by balaji-jr — DISMISSED** [review]

File: src/pages/Stream/Views/Explore/StaticLogTable.tsx (hunk 7 above)
Comment (Koustavd18) [reply src/pages/Stream/Views/Explore/StaticLogTable.tsx:216]: ok

**Review by nikhilsinhaparseable — APPROVED** [review]
good to merge

**Pull request merged**